tests/test_bots.py

```diff
@@ -26,34 +28,68 @@ def setUp(self, mock_login):
         self.falsey_values = ['no', 'false', 'f', 'n', '0']
 
     def test__init__(self):
-        bot = BaseBot()
+        bot = AbstractBotJob()
         assert bot.changed == 0
         assert bot.limit == 1
         assert bot.dry_run is True
         assert bot.logger.handlers
+        assert getattr(bot, 'console_handler', None) is not None
         assert isinstance(bot.ol, OpenLibrary)
 
     def test__str2bool_returns_true_for_truthy_input(self):
         truthy_input = self.truthy_values[random.randint(0, len(self.truthy_values) - 1)]
-        bot = BaseBot(ol=self.ol)
+        bot = AbstractBotJob(ol=self.ol)
         assert bot._str2bool(truthy_input)
 
     def test__str2bool_returns_false_for_falsey_input(self):
         falsey_input = self.falsey_values[random.randint(0, len(self.falsey_values) - 1)]
-        bot = BaseBot(ol=self.ol)
+        bot = AbstractBotJob(ol=self.ol)
         assert bot._str2bool(falsey_input) is False
 
     def test__str2bool_errors_for_non_boolean_input(self):
         non_boolean_input = random.choice(string.ascii_letters)
         while non_boolean_input in self.falsey_values or non_boolean_input in self.truthy_values:
             non_boolean_input = random.choice(string.ascii_letters)
-        bot = BaseBot(ol=self.ol)
+        bot = AbstractBotJob(ol=self.ol)
         self.assertRaises(ArgumentTypeError, bot._str2bool, non_boolean_input)
 
+    def test_dry_run_declaration_when_dry_run_is_true(self):
+        bot = AbstractBotJob(dry_run=True)
+        bot.logger.info = Mock()
+        bot.dry_run_declaration()
+        bot.logger.info.assert_called_with('dry-run is TRUE. No external modifications will be made.')  # TODO
+
+    def test_dry_run_declaration_when_dry_run_is_false(self):
+        bot = AbstractBotJob(dry_run=False)
+        bot.logger.info = Mock()
+        bot.dry_run_declaration()
+        bot.logger.info.assert_called_with('dry-run is FALSE. Permanent modifications can be made.')  # TODO
+
+    def test_process_row_with_bytecode(self):
+        random_data = list()
+        for i in range(4):
+            random_data.append(string.ascii_letters[random.randint(0, len(string.ascii_letters)-1)])
+        random_data = '\t'.join([random_data[0], random_data[1], random_data[2], random_data[3], '{"foo": "bar" }'])
```

Comment:
```suggestion
        four_random_letters = [random.choice(string.ascii_letters) for _ in range(4)]
        random_data =  "\t".join((*four_random_letters, '{"foo": "bar" }'))
```

Reply:
Punting on this one; would require a bit of testing.